### PR TITLE
Add a new test for creating a user.

### PR DESCRIPTION
**Pull request by tmyroadctfig**
I was trying to add a test to capture the problem in #19, but it seems to pass either way. Not sure if that means that the error is not bubbling up properly.

### CODE REVIEW

**Comment by OSPFNeighbour** [comment]
you can add a catch to the end of the promise but im still not sure how we tell Graphql to fail the query.

Normally i spend half my time making this fails in a nice way, this is the opposite for once.

**Comment by OSPFNeighbour** [comment]
for your test you could ask graphql to return the id or something, assuming that would be missing when the query fails.

**Comment by sdunster** [comment]
Pretty sure what you actually want to do is return a promise to mocha and have it resolve it. You don't need to add an error case unless you want to handle it. If you let the error fall through then the test will fail if there is an error. I will add an example inline.

File: server/tests/index.test.js [hunk 1]
```diff
@@ -94,4 +94,36 @@ describe('GraphQL Queries', function() {
     done();
   });
 });
+  it('should create a new user', done => {
```
Comment (sdunster) [inline server/tests/index.test.js]:
```
it('should create a new user', () => {
  // we RETURN the resulting promise for Mocha to resolve
  return self
    .test(
      JSON.stringify({
        ...
      }),
    )
    .then(res => {
      expect(res.status).toBe(200);
      expect(res.success).toBe(true);
      // you don't need to call done()
    });
    // no error case - mocha will complain if it gets one
});
```

**Comment by sdunster** [comment]
If you wanted to test the error case specifically just raise an error in your `then()` and write a normal `catch()` that verifies the error is what you expect. Because you have implemented `catch()` the error will never reach mocha so the test will pass.

btw this follows what we do in libbeacon tests. You can read more about it here: http://tobyho.com/2015/12/16/mocha-with-promises/

**Comment by OSPFNeighbour** [comment]
Err, pretty sure GraphQL isn't raising any error. As in sequaliser is
failing on a query but GraphQL just lets it and returns no error status or
message.

Same from the app, it gets no error even though half its query failed.

**Comment by sdunster** [comment]
OK so to cover this situation we need to write a test that creates a user and then runs a query to confirm the device is registered to that user.

**Comment by OSPFNeighbour** [comment]
Duplicate to #31

**Comment by sdunster** [comment]
Please close if no longer relevant.

**Pull request closed without merging**